lib/model/folder_sendrecv.go

```diff
@@ -352,7 +353,7 @@ func (f *sendReceiveFolder) processNeeded(ignores *ignore.Matcher, dbUpdateChan
 
 	select {
 	case <-f.ctx.Done():
-		return -1, nil, nil
+		return 0, nil, nil, f.ctx.Err()
```

Comment:
why 0?, why not just return the things we've processed?

Reply:
@calmh?

Reply:
So normally that's of course the case, but that's for a function that does one thing or fails. In this case I think we should legitimately return "I did 59 things, these are the results, *and then I failed* so here's also an error".